commit 80b88d9ea1b4d61b3a4ce4f5d6c6cd10ccb650ba
Author: Ozan Çağlayan <ozan@pardus.org.tr>
Date:   Thu Apr 29 14:28:56 2010 +0300

    Revert "editor: support IPv4 'disabled' method"
    
    This reverts commit baf7c01a6635a60f3f7271bb8a96d85f9a1f5e34.

diff --git a/src/connection-editor/page-ip4.c b/src/connection-editor/page-ip4.c
index 1dd4e49..8e67157 100644
--- a/src/connection-editor/page-ip4.c
+++ b/src/connection-editor/page-ip4.c
@@ -17,7 +17,7 @@
  * with this program; if not, write to the Free Software Foundation, Inc.,
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  *
- * (C) Copyright 2008 - 2010 Red Hat, Inc.
+ * (C) Copyright 2008 Red Hat, Inc.
  */
 
 #include <string.h>
@@ -33,8 +33,6 @@
 
 #include <nm-setting-connection.h>
 #include <nm-setting-ip4-config.h>
-#include <nm-setting-wired.h>
-#include <nm-setting-wireless.h>
 #include <nm-setting-gsm.h>
 #include <nm-setting-cdma.h>
 #include <nm-setting-pppoe.h>
@@ -97,7 +95,6 @@ typedef struct {
 #define IP4_METHOD_MANUAL          2
 #define IP4_METHOD_LINK_LOCAL      3
 #define IP4_METHOD_SHARED          4
-#define IP4_METHOD_DISABLED        5
 
 static void
 ip4_private_init (CEPageIP4 *self, NMConnection *connection)
@@ -178,15 +175,6 @@ ip4_private_init (CEPageIP4 *self, NMConnection *connection)
 		                    -1);
 	}
 
-	/* At the moment, Disabled is only supported for Wired & WiFi */
-	if (   priv->connection_type == NM_TYPE_SETTING_WIRED
-	    || priv->connection_type == NM_TYPE_SETTING_WIRELESS) {
-		gtk_list_store_append (priv->method_store, &iter);
-		gtk_list_store_set (priv->method_store, &iter,
-		                    METHOD_COL_NAME, _("Disabled"),
-		                    METHOD_COL_NUM, IP4_METHOD_DISABLED,
-		                    -1);
-	}
 	gtk_combo_box_set_model (priv->method, GTK_TREE_MODEL (priv->method_store));
 
 	priv->addr_label = glade_xml_get_widget (xml, "ip4_addr_label");
@@ -243,8 +231,6 @@ method_changed (GtkComboBox *combo, gpointer user_data)
 	case IP4_METHOD_MANUAL:
 		addr_enabled = dns_enabled = routes_enabled = TRUE;
 		break;
-	case IP4_METHOD_DISABLED:
-		addr_enabled = dns_enabled = dhcp_enabled = routes_enabled = FALSE;
 	default:
 		break;
 	}
@@ -331,8 +317,6 @@ populate_ui (CEPageIP4 *self)
 			method = IP4_METHOD_MANUAL;
 		else if (!strcmp (str_method, NM_SETTING_IP4_CONFIG_METHOD_SHARED))
 			method = IP4_METHOD_SHARED;
-		else if (!strcmp (str_method, NM_SETTING_IP4_CONFIG_METHOD_DISABLED))
-			method = IP4_METHOD_DISABLED;
 	}
 
 	if (method == IP4_METHOD_AUTO && nm_setting_ip4_config_get_ignore_auto_dns (setting))
@@ -888,9 +872,6 @@ ui_to_setting (CEPageIP4 *self)
 	case IP4_METHOD_SHARED:
 		method = NM_SETTING_IP4_CONFIG_METHOD_SHARED;
 		break;
-	case IP4_METHOD_DISABLED:
-		method = NM_SETTING_IP4_CONFIG_METHOD_DISABLED;
-		break;
 	case IP4_METHOD_AUTO_ADDRESSES:
 		ignore_auto_dns = TRUE;
 		/* fall through */
